src/core.cpp

```diff
@@ -63,9 +104,21 @@ void CVC::ChooseActions() {
             }
 
             //enumerate and score actions
+            std::vector<std::unique_ptr<Action>> actions = this->decision_engine_->EnumerateActions(*this, character.get());
+
             //choose one
             double choice = rand() / (RAND_MAX + 1.);
+            double sum_score = 0;
             //CVC is responsible for maintaining the lifecycle of the action
+            for(auto& action: actions) {
+                sum_score += action->GetScore();
+                if(choice < sum_score) {
+                    //keep this one action
+                    this->queued_actions_.push_back(std::move(action));
+                    //rest of the actions will go out of scope and 
```

Comment:
is this true?

Reply:
yeah, rest of actions will be destroyed.